pkg/kubelet/kubeletconfig/util/files/files_test.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package files
+
+import (
+	"fmt"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestFileExists(t *testing.T) {
+	fs := utilfs.NewFakeFs()
+
+	testcases := []struct {
+		name         string // name of test case
+		filePath     string // file name to be created
+		needsCreated bool   // if should create the file before test
+		isFile       bool   // is the fileName a file or folder
+		expectedBool bool   // expected result in bool
+	}{
+		{
+			name:         "not exists",
+			filePath:     "/tmp/FileExists",
+			needsCreated: false,
+			isFile:       false,
+			expectedBool: false,
+		},
+		{
+			name:         "exists as folder",
+			filePath:     "/tmp/FileExists",
+			needsCreated: true,
+			isFile:       false,
+			expectedBool: true,
+		},
+		{
+			name:         "exists as file",
+			filePath:     "/tmp/FileExists/file",
+			needsCreated: true,
+			isFile:       true,
+			expectedBool: true,
+		},
+	}
+
+	for _, testcase := range testcases {
+		var err error
+		t.Run(testcase.name, func(t *testing.T) {
+			if testcase.needsCreated {
+				if testcase.isFile {
+					// create as file
+					_, err = fs.TempFile(testcase.filePath, "")
+				} else {
+					// create as folder
+					err = fs.MkdirAll(testcase.filePath, defaultPerm)
+				}
+			}
+
+			if err != nil {
+				t.Errorf("unexpected error in creating file %s, err: %v", testcase.filePath, err)
```

Comment:
@mtaufen Hi, I found that utilfs.NewFakeFs() can not satisfy the needs for my test cases. Cause the Filesystem created by utilfs.NewFakeFs() can not distinguish between file and dir, which I discover that utilfs.NewFakeFs().Stat().Mode().IsRegular() did not match the behavior of os package.
Do you think it's ok to mock another Filesystem leveraging with os package? And I will remove the temporary files and dirs after test cases, otherwise seems there is no FakeFilesystem I can use to accomplish these test cases.

Reply:
You can just use the DefaultFs for your test, just be sure to generate a temp dir in /tmp and clean it up after your test.